specs/interop/token-bridging.md

```diff
@@ -2,96 +2,298 @@
 
 <!-- START doctoc generated TOC please keep comment here to allow auto update -->
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
-**Table of Contents**
 
-- [Interface](#interface)
+- [Overview](#overview)
 - [Implementation](#implementation)
-- [L1 Native Tokens](#l1-native-tokens)
-- [Cross Chain `transferFrom`](#cross-chain-transferfrom)
-- [Factory](#factory)
-- [Upgrading Existing `OptimismMintableERC20Token`s](#upgrading-existing-optimismmintableerc20tokens)
+- [Invariants](#invariants)
+- [Deployment and migrations](#deployment-and-migrations)
+  - [Factory](#factory)
+  - [Migration to the Standard](#migration-to-the-standard)
+- [Future Considerations](#future-considerations)
+  - [Cross Chain `transferFrom`](#cross-chain-transferfrom)
+    - [Approvals](#approvals)
+    - [`remoteTransferFrom()`](#remotetransferfrom)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-Unified ERC20 token bridging built on top of the messaging protocol
-can enable tokens to be fungible and movable across the superchain.
-An issue with bridging is that if the security model is not standardized,
-then the bridged assets are not fungible with each other depending on
-which bridge solution was used to facilitate the transfer between
-domains.
+## Overview
 
-## Interface
+Without a standardized security model, bridged assets may not be fungible with each other. The `SuperchainERC20` unifies ERC20 token bridging to make it fungible across the Superchain. It builds on top of the messaging protocol, as the most trust minimized bridging solution.
 
-The `ISuperchainERC20` interface creates a standard way to send tokens
-between domains. In contrast to [xERC20][xerc20] where the user calls
-a bridge that has `mint` and `burn` privileges on the cross chain enabled
-tokens, the user instead calls methods on the token contract directly.
+Unlike other standards, such as [xERC20](https://www.xerc20.com/), where users interact with a bridge possessing mint and burn privileges on cross-chain enabled tokens, this approach allows users to call methods directly on the token contract.
 
-[xerc20]: https://www.xerc20.com/
+## Implementation
 
-```solidity
-interface ISuperchainERC20 {
-  function bridgeERC20(uint256 amount, uint256 chainId) external;
-  function bridgeERC20To(address to, uint256 amount, uint256 chainId) external;
-  function finalizeBridgeERC20(address to, uint256 amount) external;
-}
-```
+The standard will build on top of ERC20 and include the following five external functions:
 
-## Implementation
+- `sendERC20(uint256 amount, uint256 chainId)(bool)`
+- `sendERC20(address to, uint256 amount, uint256 chainId)(bool)`
+- `sendERC20(address to, uint256 amount, uint256 chainId, bytes data)(bool)`
+- `finalizeSendERC20(address to, uint256 amount)`
+- `finalizeSendERC20(address to, uint256 amount, bytes data)`
```

Comment:
small nit: can we modify the finalize functions to be `relayERC20` to match the interface of the messenger? just to keep naming consistent

https://github.com/ethereum-optimism/specs/blob/main/specs/interop/predeploys.md#relaymessage-invariants

Reply:
Also thoughts on just having one `relayERC20` function that optionally makes a `CALL` to `to` if `data.length > 0`? Seems like a simplification that we could make to make the interface smaller. We could either keep both `sendERC20` functions or only have the one with `data`, I lean towards keeping both so it feels like `transfer` but open to keeping both to keep the standard smaller

Reply:
I really like the idea of simplifying both `relayERC20` and `sendERC20` to a single function. 
- I don't feel the analogy with `transfer` is strong enough to keep both, as it's a whole different action. 
- UX wise, we should trust the integrations and keep the standard as lean as possible.
- Gas wise, there will be an advantage for keeping the functions separate, but its only a very small check.

Reply:
We are thinking about changing the - `sendERC20(address to, uint256 amount, uint256 chainId, bytes data)` to include a `target` input. In this way, we could decouple the call from the `to`. That might be an argument for of keeping both implementations.

Reply:
moved into single `sendERC20` and `relayERC20` in the latest commit. We can keep the discussion open just in case.